ide/lsp.client/src/org/netbeans/modules/lsp/client/debugger/breakpoints/DAPBreakpointConvertor.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.lsp.client.debugger.breakpoints;
+
+import org.netbeans.api.debugger.Breakpoint;
+import org.netbeans.modules.lsp.client.debugger.spi.BreakpointConvertor;
+import org.openide.util.lookup.ServiceProvider;
+
+@ServiceProvider(service=BreakpointConvertor.class)
+public class DAPBreakpointConvertor implements BreakpointConvertor {
+
+    @Override
+    public void covert(Breakpoint b, ConvertedBreakpointConsumer breakpointConsumer) {
+        if (b instanceof DAPLineBreakpoint lb) {
+            breakpointConsumer.lineBreakpoint("file://" + lb.getFilePath(), lb.getLineNumber(), lb.getCondition());
```

Comment:
This looks broken. What is in `lb.getFilePath`? Is this a local path? Then the code presented above using `java.io.File` can be used to convert, if it is something different, what is it?

Reply:
I agree. This looked wrong. The whole duality of breakpoint having a `FileObject` and `filePath` was strange thing to copy&paste. As far as I can say, since https://github.com/apache/netbeans/pull/8098#issuecomment-2571615395 it is fixed and this code is using more sane `lb.getFileObject().toURI()`.